### DISCUSSION TITLE
Pessoal na opinião de vcs qual a melhor IDE Atom ou VSCode? e pq?

### DISCUSSION THREAD

**Discussion by ghost** [frontendbr/forum#1240] [Perguntas e Respostas]
Pessoal na opinião de vcs qual a melhor IDE Atom ou VSCode? e pq?

**Comment by vitor-mendrone-deel** [comment]
@Bruno413 Tomei a liberdade de editar o conteúdo da issue porque o anterior era só o que sugerimos na template, ok?

**Comment by ghost** [comment]
ok :+1:

**Comment by MarcoWorms** [comment]
O VSCode pra mim funciona bem parecido com o atom apesar de eu não usar nenhum dos 2, mas dado que o VSCode é da microsoft e ela comprou o Github (que fazia o atom) creio que a longo prazo o VSCode tem uma vantagem aí

**Comment by ghost** [comment]
Verdade, tinha esquecido que a Microsoft comprou o github, o Atom pertence ao github logo também é da microsft, será que vão manter as 2 IDE?

**Comment by alexandreramosdev** [comment]
Acredito que manterá sim!  pois quem mantem a IDE são as comunidades!  Uma dela so sumira caso a comunidade não queira mais assim como tantas outras IDE

**Comment by ghost** [comment]
pode ser, as 2 são bem parecidas e tem o mesmo dono poderiam se juntar

**Comment by tcelestino** [comment]
> pode ser, as 2 são bem parecidas e tem o mesmo dono poderiam se juntar

Ao meu ver, o Atom e o VSCode possuem propostas diferentes, mesmo a MS apostando na integração com o Github no VSCode, acho que o Atom está mais integrado. Sem contar que... parece que o Atom é bem pior em performance em relação ao VSCode, devido a isso acho bem "difícil" a MS juntar o core dos dois. Pode ser que um fique mais popular que o outro (o que já vem acontecendo hoje em dia).

**Comment by leandrocunha** [comment]
Antes do VSCode eu era muito fã do Atom, só que de fato usei bem pouco pq ele comia muita memoria do meu macbook e fica inviavel desenvolver com isso usei por muito tempo o Sublime.

Hoje depois de muita resistencia mudei pro VSCode e realmente o trabalho que os caras estão fazendo é incrível, em termos de performance, intelisense, integrações e extensões o VSCode ganha fácil dos outros.

Mas tb temos que levar em conta que o Atom e o Sublime são editores e o VSCode é uma IDE mesmo.

**Comment by phenriqueleao** [comment]
> Sem contar que... parece que o Atom é bem pior em performance em relação ao VSCode, devido a isso acho bem "difícil" a MS juntar o core dos dois.

Esse é o motivo de eu usar o VSCode, tentei usar Atom e achei a performance bem ruim no meu PC.

**Comment by felipefialho** [comment]
Acho o VSCode simplesmente espetacular, mantém as qualidades do Sublime: Simplicidade e performance, e adiciona features incríveis como o terminal integrado.

Além disso a biblioteca de plugins é enorme e com atualizações constantes.

Dificilmente trocaria hoje.

**Comment by thismarcoantonio** [comment]
Acredito que o que eu acho melhor no VSCode é a facilidade de personalizar tudo, javascript é maravilhoso <3

Outra coisa que me chama bastante atenção é a constante melhoria do mesmo, claro, tem algumas coisas que sempre vão desagradar, mas sempre dá pra desativar, só pesquisar um pouquinho (como aquela recente adição onde mostrava os imports não utilizados em uma opacidade mais leve).

E essa melhoria visual na barra superior do windows me fez até voltar a desenvolver em casa:
![vscode windows](https://user-images.githubusercontent.com/25869417/47193094-7a123900-d327-11e8-841c-ba2ddf8c7ecb.png)

**Reply by FallzOfficial** [reply]
Qual tema você usa no seu VSCode?

**Reply by ojvribeiro** [reply]
Acredito que o syntax highlight seja a mesma que uso: [Material Syntax Dark](https://marketplace.visualstudio.com/items?itemName=woodywoodsta.vscode-material-syntax-dark)

**Comment by didiraja** [comment]
VS Code de longe, constantemente atualizado além de um monte de função bacana já padrão (integração com git, lint, debug mode e customização extensa)

**Comment by aloisogomes** [comment]
VScode é superior!  Eu era um franco defensor do atom, mas o ultimo vem enfrentando alguns problemas, ele consome uma memoria absurda se vc tem um note ou pc comum ele vai travar e vc vai se sentir frustrado programando.. (O terror de teclar e 5 segundos depois aparecer cada caracter) os plugins estao desatualizados alguns dao bug ao instalar..

O VScode tem mais plugins (que eles chamam de extensoes) e parece q tem uma comunidade ativa mantendo eles. Eu tenho um note meia boca e o VScode abriu tao rapido e no jeito q eu achei realmente que ele nao funcionava rsrs

Ah mas eu sou super acostumado com as keybindings do atom. Acredita que o VScode em preferencias tem uma area para importar as keybindings do atom? Sublime tmb! 

Olha se vc ta lendo isso é pq ta em duvida, se aceita a opiniao desse developer aqui, o VScode é a aposta no cavalo certo. Esses caras enfiaram o pé no acelerador essa ide realmente faz parecer facil programar (nao deixem os clientes de vcs lerem isso)

Quanto ao Atom, obrigado por tudo, mas passarei a ver vc apenas pelo retrovisor. Abraço a tds.

**Reply by ojvribeiro** [reply]
Inclusive, sou um ex usuário do Atom e migrei há alguns anos pro VSCode, então importei todas as teclas de atalho e sou eternamente feliz. Até usei o mesmo tema que usava lá (Material Syntax Highlight Dark).

O VSCode é um monstro com tanto recurso que é impossível chamar de apenas um mero editor de código.

**Comment by mazoqui** [comment]
Pra mim, um dos pontos negativos do Atom é depuração de código python. 
Com relação ao consumo de memoria, a diferença não foi grande no teste que fiz:
Duas VMs Windows 10 (configuração limpa) - Instalei o basico das IDEs (sem plugins ou extensoes). 
No mesmo projeto testei o consumo pra 1, 10 e 20 abas abertas: ATOM: 468Mb, 501Mb e 522Mb
e VScode: 430Mb, 469Mb e 527Mb. O consumo de CPU tbem ficou baixo tbem 1~2% em ambas
Mas acho o VScode mais rapido na interação.

**Comment by Magoninho** [comment]
Eu uso o VSCode porque o Atom consome muita RAM. Infelizmente vivemos numa era onde a maioria dos aplicativos optaram por usar Electron e consomem muita RAM. Mas, de algum jeito o VSCode fez alguma magia pra deixar o aplicativo mais leve que o Atom mesmo sendo em Electron (apesar de ainda consumir por volta de 350mb). Apesar do Atom ser uma boa tentativa de substituir o Sublime, ele falhou em seu desenvolvimento por consumir mais de 1gb da minha RAM (sem extensões instaladas, apenas com uma pasta de um projeto). No final eu desejava que tudo fosse feito em C ou C++, aí teriamos performance absurda 👍🏻

**Comment by leandrocrs** [comment]
Só mudo do VSCode se for pra uma IDE completa ou pra um editor tão completo quanto mas com desempenho melhor. Atom fora de cogitação. Ele é bom, mas acho que esse tema já foi vencido a pelo menos uns 2 anos atrás. VSCode é muito superior.

**Comment by CacioDavi** [comment]
atom e vs code são os melhores porém as extensões do vs code tornam ele melhor que o atom

**Comment by rodrigoreis** [comment]
Não são IDEs mas sim editores de código.

**Reply by CacioDavi** [reply]
IDE: Ambiente de desenvolvimento integrado
IDE, do inglês Integrated Development Environment ou Ambiente de Desenvolvimento Integrado, é um programa de computador que reúne características e ferramentas de apoio ao desenvolvimento de software com o objetivo de agilizar este processo.

**Reply by includebrasil** [reply]
> Não são IDEs mas sim editores de código.

Correto, até hoje esse povo não aprendeu! hehehehe

**Reply by ojvribeiro** [reply]
> > Não são IDEs mas sim editores de código.
> 
> Correto, até hoje esse povo não aprendeu! hehehehe

O que uma IDE tem que o VSCode não tem?

**Reply by includebrasil** [reply]
> > > Não são IDEs mas sim editores de código.
> > 
> > 
> > Correto, até hoje esse povo não aprendeu! hehehehe
> 
> O que uma IDE tem que o VSCode não tem?

A começar que no site do VSCode diz: O Visual Studio Code é um editor de código-fonte leve... o resto vc descobre trabalhando com uma IDE e com o VSCode.

**Reply by aloisogomes** [reply]
Acho graça quando alguém vem aqui com a chance de esclarecer a diferença sobre diferença de IDE ou editor de código com integração de Plugins como o VScode e responde algo genérico como veja usando.. só posso pensar que ela sabe muito do assunto ou não sabe nada.

O VScode de fato é um editor de código que com plugins pode te ajudar a ter controle e informação sobre a sintaxe da linguagem que você está usando fazendo com o que seu desenvolvimento esteja próximo do adequado para sua aplicação. Embora vc tenha a impressão que ele esteja fazendo compilações e debugs via análise de código ele não o faz. Eis aí a grande diferença

Uma ide, vamos pensar por exemplo no Android Studio, é de fato um ecossistema totalmente voltado para a entrega da plataforma que você está fazendo ele não somente analisa a sintaxe de seu código mas já te ajuda com as compilações e depuracoes em tempo real para a plataforma inclusive mostrando imediatamente os problemas e te ajudando como resolver. 

Claro que é uma breve explicação bastante rasa aqui. Diria que poderia ser dedicada uma obra sobre isso. Mas acho mais maneiro do que dar de ombros

**Reply by rodrigoreis** [reply]
Primeiramente, você sabe se a pessoa quer fazer essa explicação? Sua atitude foi louvável em definir, mas foi péssima em tentar impor o que uma pessoa deve ou não fazer num fórum aberto e sem regras.

**Reply by aloisogomes** [reply]
Perdoe-me pelas palavras. Jamais quis ofender ou impor algo. Vejo um fórum mais do que uma terra de ninguém e sem leis. Meu otimismo me leva a enxergar como uma fonte de conhecimento para as próximas gerações. Uma resposta interessada representa mais que uma resposta que não coloca luz sobre nada. Todos temos o direito de cooperar como quiser. Mas não acho que tenha prestimo algum dar uma opinião que não agrega valor. 

Desculpem aos colegas que não se sentiram bem com isso.

**Reply by nazarepiedady** [reply]
Os dois editores de códigos da comparação são lentos,
se não estiver escrevendo com SSD e memória RAM de pelo menos 8GB,
pode começar a costumar-se com travagens no sistema operacional.

As velhas promessas de desempenho otimizado da Microsoft é pura propaganda,
pelo menos até ao momento.

**Reply by rodrigoreis** [reply]
> Perdoe-me pelas palavras. Jamais quis ofender ou impor algo. Vejo um fórum mais do que uma terra de ninguém e sem leis. Meu otimismo me leva a enxergar como uma fonte de conhecimento para as próximas gerações. Uma resposta interessada representa mais que uma resposta que não coloca luz sobre nada. Todos temos o direito de cooperar como quiser. Mas não acho que tenha prestimo algum dar uma opinião que não agrega valor.
> 
> Desculpem aos colegas que não se sentiram bem com isso.

Amigo, sem prolongar demais essa discussão que começa, para mim, a ficar sem sentido, mas gostaria de saber como você define que uma opinião não agrega valor? Talvez para você apenas. O colega acima deu a opinião se você não gostou ou achou incompleta, complemente sem desmerecer o mesmo, penso assim.

**Reply by felquis** [reply]
Verdade desviou muito do assunto. @rodrigoreis recomendo criar uma discussão focada na temática da sua pergunta assim a gente separa um pouco os assuntos.

💡  Também vamos tentar focar nas partes boas dos comentários dos coleguinhas, pois se for destacar apenas a parte ruim todos nós sairemos perdendo.

Como o tópico já tem 4 anos aqui no fórum e a pessoa que criou originalmente já não está mais aqui, (portanto ela não tem interesse legítimo na continuação do tópico) essa discussão poderá ser fechada de acordo com descrito em https://github.com/frontendbr/forum/discussions/2291

**Comment by nazarepiedady** [comment]
Existem opções de melhor desempenho que estas duas da comparação,
por exemplo, Vim, Sublime Text, Emacs.
